common/cmake/devices.cmake

```diff
@@ -993,16 +999,16 @@ function(ADD_FPGA_TARGET)
   endforeach()
 
   if(NOT ${ADD_FPGA_TARGET_NO_SYNTHESIS})
-    set(COMPLETE_YOSYS_SYNTH_SCRIPT "tcl ${YOSYS_SYNTH_SCRIPT}")
+    set(COMPLETE_YOSYS_SYNTH_SCRIPT "plugin -i xdc; plugin -i fasm; tcl ${YOSYS_SYNTH_SCRIPT}")
 
     set(OUT_JSON_SYNTH ${OUT_LOCAL}/${TOP}_synth.json)
     set(OUT_JSON_SYNTH_REL ${OUT_LOCAL_REL}/${TOP}_synth.json)
     set(OUT_JSON ${OUT_LOCAL}/${TOP}.json)
     set(OUT_JSON_REL ${OUT_LOCAL_REL}/${TOP}.json)
 
     add_custom_command(
-      OUTPUT ${OUT_JSON_SYNTH} ${OUT_SYNTH_V}
-      DEPENDS ${SOURCE_FILES} ${SOURCE_FILES_DEPS}
+      OUTPUT ${OUT_JSON_SYNTH} ${OUT_SYNTH_V} ${OUT_FASM_EXTRA}
```

Comment:
Because `OUT_FASM_EXTRA` is not always generated, this `add_custom_command` is fragile and may break.  Only add the additional output if this will be actually be generated?

Reply:
One option here is to add a third command to the `add_custom_command` to touch `OUT_FASM_EXTRA` so it always exists, even if the yosys script doesn't write anything.